code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm

```diff
@@ -0,0 +1,826 @@
+		qdel(src)
+	else
+		add_language("TerrorSpider")
+		add_language("Galactic Common")
+		default_language = all_languages["TerrorSpider"]
+
+		name += " ([rand(1, 1000)])"
+		msg_terrorspiders("[src] has grown in [get_area(src)].")
+		if (name_usealtnames)
+			name = pick(altnames)
+		if(istype(get_area(src), /area/awaycontent) || istype(get_area(src), /area/awaymission/))
+			spider_awaymission = 1
+			if (spider_tier >= 3)
+				ai_ventcrawls = 0 // means that pre-spawned bosses on away maps won't ventcrawl. Necessary to keep prince/mother in one place.
+			if (istype(get_area(src), /area/awaymission/UO71)) // if we are playing the away mission with our special spiders...
+				spider_uo71 = 1
+				if (world.time < 600)
+					// spider is spawning in the first 60 seconds of the round, asssume its an awaymission spider placed there with the intention of staying there.
+					ai_ventcrawls = 0
+					spider_placed = 1
+					wander = 0
+			ts_count_alive_awaymission++
+		else
+			ts_count_alive_station++
+		// after 30 seconds, assuming nobody took control of it yet, offer it to ghosts.
+		spawn(150) // deciseconds!
+			CheckFaction()
+		spawn(300) // deciseconds!
+			if (spider_awaymission)
+				// never announce awaymission spiders.
+			else if (ckey)
+				notify_ghosts("[src] has appeared in [get_area(src)]. (already player-controlled)")
+			else if (ai_playercontrol_allowingeneral && ai_playercontrol_allowtype)
+				notify_ghosts("[src] has appeared in [get_area(src)]. <a href=?src=\ref[src];activate=1>(Click to control)</a>")
+
+
+
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/Life()
+	if (stat != DEAD)
+		if (degenerate > 0)
+			adjustToxLoss(rand(1,10))
+		if (regen_points < regen_points_max)
+			regen_points += regen_points_per_tick
+		if ((bruteloss > 0) || (fireloss > 0))
+			if (regen_points > regen_points_per_hp)
+				if (bruteloss > 0)
+					adjustBruteLoss(-1)
+					regen_points -= regen_points_per_hp
+				else if (fireloss > 0)
+					adjustFireLoss(-1)
+					regen_points -= regen_points_per_hp
+		if (prob(5))
+			CheckFaction()
+	if (stat == DEAD)
+		if (prob(2))
+			// 2% chance every cycle to decompose
+			visible_message("<span class='notice'>\the dead body of the [src] decomposes!</span>")
+			gib()
+	..()
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/death(gibbed)
+	if (!gibbed)
+		msg_terrorspiders("[src] has died in [get_area(src)].")
+		//if (!ckey && spider_tier < 3)
+		//	say(pick("Mistresssss will end you...", "Doom waitssss... for you...","She comessssss for your flesh..."))
+	if (!hasdroppedloot)
+		hasdroppedloot = 1
+		if (ts_count_dead == 0)
+			visible_message("<span class='userdanger'>The Terrors have awoken!</span>")
+		ts_count_dead++
+		ts_death_last = world.time
+		if (spider_awaymission)
+			ts_count_alive_awaymission--
+		else
+			ts_count_alive_station--
+	..()
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/handle_automated_action()
```

Comment:
This has been refactored to remove the spider subtype checks, and their associated conditionals.